.github/workflows/azure-e2e-test.yml

```diff
@@ -0,0 +1,151 @@
+        pushd ../
+        git clone https://github.com/kata-containers/kata-containers -b CCv0
+        popd
+
+    - name: Build binaries
+      env:
+        GOPATH: /home/runner/go
+      run: |
+        pushd azure/image
+        make binaries
+        popd
+
+    - uses: azure/login@v1
+      name: 'Az CLI login'
+      with:
+        client-id: ${{ secrets.AZURE_CLIENT_ID }}
+        subscription-id: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
+        tenant-id: ${{ secrets.AZURE_TENANT_ID }}
+
+    - name: Create podvm image
+      id: create-podvm-image
+      env:
+        PKR_VAR_subscription_id: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
+        PKR_VAR_resource_group: ${{ secrets.AZURE_RESOURCE_GROUP }}
+        PKR_VAR_location: ${{ secrets.AZURE_REGION }}
+        PKR_VAR_az_image_name: ${{ env.PODVM_IMAGE_NAME }}
+        PKR_VAR_vm_size: ${{ env.VM_SIZE }}
+        PKR_VAR_ssh_username: ${{ env.SSH_USERNAME }}
+        PKR_VAR_az_gallery_name: ${{ env.PODVM_GALLERY_NAME }}
+        PKR_VAR_az_gallery_image_name: ${{ env.PODVM_IMAGE_DEF_NAME }}
+        PKR_VAR_az_gallery_image_version: ${{ env.PODVM_IMAGE_VERSION }}
+        PKR_VAR_use_azure_cli_auth: "true"
+        GOPATH: /home/runner/go
+        PODVM_DISTRO: "ubuntu"
+      run: |
+        pushd azure/image
+        make image
+        popd
+
+  run-e2e-test:
+    runs-on: ubuntu-latest
+    needs:
+      - build-podvm-image
+    steps:
+    - uses: actions/checkout@v2
+
+    - uses: azure/login@v1
+      name: 'Az CLI login'
+      with:
+        client-id: ${{ secrets.AZURE_CLIENT_ID }}
+        subscription-id: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
+        tenant-id: ${{ secrets.AZURE_TENANT_ID }}
+
+    - name: Create provisioner file
+      env:
+        AZURE_SUBSCRIPTION_ID: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
+        AZURE_CLIENT_ID: ${{ secrets.AZURE_CLIENT_ID }}
+        AZURE_TENANT_ID: ${{ secrets.AZURE_TENANT_ID }}
+        RESOURCE_GROUP_NAME: ${{ secrets.AZURE_RESOURCE_GROUP }}
+        CLUSTER_NAME: "e2e-test-cluster"
+        LOCATION: ${{ secrets.AZURE_REGION }}
+        SSH_KEY_ID: "id_rsa.pub"
+        CLOUD_PROVIDER: "azure"
+        PODVM_IMAGE_NAME: ${{ env.PODVM_IMAGE_NAME }}
+        TEST_E2E_CREATE_RG: "no"
+      run: |
+        cat << EOF > /tmp/provision_azure.properties
+          AZURE_SUBSCRIPTION_ID="$AZURE_SUBSCRIPTION_ID"
+          AZURE_CLIENT_ID="$AZURE_CLIENT_ID"
+          AZURE_CLIENT_SECRET="$AZURE_CLIENT_SECRET"
+          AZURE_TENANT_ID="$AZURE_TENANT_ID"
+          RESOURCE_GROUP_NAME="$RESOURCE_GROUP_NAME"
+          CLUSTER_NAME="$CLUSTER_NAME"
+          LOCATION="$LOCATION"
+          SSH_KEY_ID="$SSH_KEY_ID"
+          CLOUD_PROVIDER="$CLOUD_PROVIDER"
+          AZURE_IMAGE_ID="/subscriptions/$AZURE_SUBSCRIPTION_ID/resourceGroups/$RESOURCE_GROUP_NAME/providers/Microsoft.Compute/images/$PODVM_IMAGE_NAME"
+          IS_CI_MANAGED_CLUSTER="true"
+          AZURE_CLI_AUTH="true"
+          MANAGED_IDENTITY_NAME="ghactions_user3"
```

Comment:
nit: can we use a more descriptive name?

Reply:
These envs name are from go env used in provisioner. If we need to change these we need to change provisioner code. I will make different change if these names need to be changed.